spec/cc/engine/analyzers/go/main_spec.rb

```diff
@@ -0,0 +1,180 @@
+        expect(json["categories"]).to eq(["Duplication"])
+        expect(json["location"]).to eq({
+          "path" => "foo.go",
+          "lines" => { "begin" => 9, "end" => 12 },
+        })
+        expect(json["remediation_points"]).to eq(380_000)
+        expect(json["other_locations"]).to eq([
+          {"path" => "foo.go", "lines" => { "begin" => 14, "end" => 17} },
+          {"path" => "foo.go", "lines" => { "begin" => 19, "end" => 22} },
+        ])
+        expect(json["content"]["body"]).to match(/This issue has a mass of 32/)
+        expect(json["fingerprint"]).to eq("729e7221b4530916ed63cfb6f4a3fe90")
+        expect(json["severity"]).to eq(CC::Engine::Analyzers::Base::MINOR)
+      end
+
+      it "prints an issue for similar code" do
+        create_source_file("foo.go", <<-EOGO)
+          package main
+
+          import "fmt"
+
+          func add(x int, y int) int {
+          	return x + y
+            fmt.Println("Add some stuff!")
+          }
+
+          func add(x int, y int) int {
+            return x + y
+            fmt.Println("Add all the stuff!")
+          }
+
+          func main() {
+          	fmt.Println(add(42, 13))
+            fmt.Println(add(44, 15))
+          }
+        EOGO
+
+        issues = run_engine(engine_conf).strip.split("\0")
+        result = issues.first.strip
+        json = JSON.parse(result)
+
+        expect(json["type"]).to eq("issue")
+        expect(json["check_name"]).to eq("similar-code")
+        expect(json["description"]).to eq("Similar blocks of code found in 2 locations. Consider refactoring.")
+        expect(json["categories"]).to eq(["Duplication"])
+        expect(json["location"]).to eq({
+          "path" => "foo.go",
+          "lines" => { "begin" => 5, "end" => 8 },
+          })
+        expect(json["remediation_points"]).to eq(900_000)
+        expect(json["other_locations"]).to eq([
+          {"path" => "foo.go", "lines" => { "begin" => 10, "end" => 13} },
+        ])
+        expect(json["content"]["body"]).to match /This issue has a mass of 45/
+        expect(json["fingerprint"]).to eq("f1551c88ceadf1241f6a0c92cce82413")
+        expect(json["severity"]).to eq(CC::Engine::Analyzers::Base::MAJOR)
+      end
+
+      it "outputs a warning for unprocessable errors" do
+        create_source_file("foo.go", <<-EOGO)
+          ---
+        EOGO
+
+        expect(CC.logger).to receive(:warn).with(/Response status: 422/)
+        expect(CC.logger).to receive(:warn).with(/Skipping/)
+        run_engine(engine_conf)
+      end
+
+      it "ignores import declarations" do
+        create_source_file("foo.go", <<-EOGO)
+          package main
+
+          import "fmt"
+          import "fmt"
+          import "fmt"
+
+          func main() {
+            fmt.Println("Hello!")
+          }
+        EOGO
+
+        issues = run_engine(engine_conf).strip.split("\0")
+        expect(issues).to be_empty
+      end
```

Comment:
I might suggest we modify this test to use two separate source files (ex. `foo.go` and `bar.go`) since this example does not compile:

```
tmp/sandbox319209057/main.go:4:18: fmt redeclared as imported package name
	previous declaration at tmp/sandbox319209057/main.go:3:18
```

FYI: you can easily try out these examples at [The Go Playground](https://play.golang.org/).